stresstests/src/test/java/org/neo4j/causalclustering/stresstests/StartStopLoad.java

```diff
@@ -21,46 +21,45 @@
 
 import java.io.File;
 import java.io.PrintStream;
-import java.util.concurrent.locks.LockSupport;
-import java.util.function.BooleanSupplier;
 
 import org.neo4j.causalclustering.catchup.storecopy.CopiedStoreRecovery;
 import org.neo4j.causalclustering.catchup.storecopy.TemporaryStoreDirectory;
-import org.neo4j.causalclustering.discovery.Cluster;
 import org.neo4j.causalclustering.discovery.ClusterMember;
 import org.neo4j.consistency.ConsistencyCheckService;
 import org.neo4j.io.fs.FileSystemAbstraction;
 import org.neo4j.io.pagecache.PageCache;
 import org.neo4j.io.pagecache.impl.muninn.StandalonePageCacheFactory;
 import org.neo4j.kernel.configuration.Config;
 import org.neo4j.kernel.extension.KernelExtensions;
+import org.neo4j.logging.Log;
 
 import static org.neo4j.consistency.ConsistencyCheckTool.runConsistencyCheckTool;
 import static org.neo4j.io.NullOutputStream.NULL_OUTPUT_STREAM;
 
-class StartStopLoad extends RepeatUntilOnSelectedMemberCallable
+class StartStopLoad extends RepeatOnRandomMember
 {
     private final FileSystemAbstraction fs;
     private final PageCache pageCache;
+    private final Log log;
 
-    StartStopLoad( FileSystemAbstraction fs, PageCache pageCache, BooleanSupplier keepGoing, Runnable onFailure,
-            Cluster cluster, int numberOfCores, int numberOfEdges )
+    StartStopLoad( Control control, Resources resources )
     {
-        super( keepGoing, onFailure, cluster, numberOfCores, numberOfEdges );
-        this.fs = fs;
-        this.pageCache = pageCache;
+        super( control, resources );
+        this.fs = resources.fileSystem();
+        this.pageCache = resources.pageCache();
+        this.log = resources.logProvider().getLog( getClass() );
     }
 
     @Override
-    protected void doWorkOnMember( boolean isCore, int id )
+    protected void doWorkOnMember( ClusterMember member ) throws InterruptedException
     {
-        ClusterMember member = isCore ? cluster.getCoreMemberById( id ) : cluster.getReadReplicaById( id );
         File storeDir = member.database().getStoreDir();
-        KernelExtensions kernelExtensions =
-                member.database().getDependencyResolver().resolveDependency( KernelExtensions.class );
+        KernelExtensions kernelExtensions = member.database().getDependencyResolver().resolveDependency( KernelExtensions.class );
+        log.info( "Stopping: " + member );
         member.shutdown();
         assertStoreConsistent( storeDir, kernelExtensions );
-        LockSupport.parkNanos( 5_000_000_000L );
+        Thread.sleep( 5000 );
```

Comment:
Is this sleep strictly necessary? Could the consistency check be wrapped in a completable future or something? I suppose with workloads taking this long 5 seconds isn't that big of a deal though.

Reply:
The sleep has nothing to do with the consistency check really. It is just a rate-limiter for the speed at which we start and stop instances. I could imagine perhaps making the sleep-time randomized to hit various cases due to the shifts in timing in the stress testing. I think the consistency check might also be made randomized (e.g. happens on scale from 0 to 100% of the time).

Reply:
Ah I see - my mistake.